force_wfmanager/tests/test_wfmanager_review_task.py

```diff
@@ -59,74 +69,100 @@ def test_save_project_failure(self):
             mock_error.assert_called_with(
                 None,
-                'Cannot save in the requested file:\n\nOUPS',
-                'Error when saving the project'
+                "Cannot save in the requested file:\n\nOUPS",
+                "Error when saving the project",
+            )
+
+        error = ValueError("some wrong value")
+        mock_open = mock.mock_open()
+        mock_open.side_effect = error
+        with mock.patch(
+            RESULTS_FILE_DIALOG_PATH
+        ) as mock_file_dialog, mock.patch(
+            RESULTS_FILE_OPEN_PATH, mock_open, create=True
+        ), mock.patch(
+            RESULTS_ERROR_PATH
+        ) as mock_error:
+            mock_file_dialog.side_effect = mock_dialog(FileDialog, OK)
+            mock_error.side_effect = mock_return_args
+
+            self.review_task.save_project_as()
+
+            self.assertTrue(mock_open.called)
+            mock_error.assert_called_with(
+                None,
+                f"Cannot save the Project:\n\n{error}",
+                "Error when saving the project",
             )
 
     def test_default_data_views(self):
         # Test of the data view selection feature within the task.
 
         # A single default plot type
         self.assertEqual(
-            len(self.review_task.central_pane.available_data_views), 1)
+            len(self.review_task.central_pane.available_data_views), 1
+        )
 
         # Initial state
         self.assertEqual(
-            self.review_task.central_pane.data_view_selection, Plot)
-        self.assertIsInstance(
-            self.review_task.central_pane.data_view, Plot)
+            self.review_task.central_pane.data_view_selection, Plot
+        )
+        self.assertIsInstance(self.review_task.central_pane.data_view, Plot)
 
         # The "change" button needs to be fired to populate the descriptions
         with mock.patch(DATA_VIEW_PANE_EDIT_TRAITS_PATH) as mock_edit_traits:
             self.review_task.central_pane.change_view = True
             mock_edit_traits.assert_called_with(view="selection_changer")
         self.assertIn(
             "Plot with colormap (force_wfmanager.ui.review.plot.Plot)",
-            self.review_task.central_pane.data_view_descriptions.values()
+            self.review_task.central_pane.data_view_descriptions.values(),
         )
 
 
 class TestWFManagerTasksWithPlugins(GuiTestAssistant, TestCase):
     def setUp(self):
         super(TestWFManagerTasksWithPlugins, self).setUp()
         _, self.review_task = get_probe_wfmanager_tasks(
-            wf_manager=DummyWfManagerWithPlugins())
+            wf_manager=DummyWfManagerWithPlugins()
+        )
 
     def test_discover_data_views(self):
         # One default plot type plus three contributed
         self.assertEqual(
-            len(self.review_task.central_pane.available_data_views), 4)
+            len(self.review_task.central_pane.available_data_views), 4
+        )
```

Comment:
(TST: should this be `4, len(self.review_task.central_pane.available_data_views)` instead?)